github/client.go

```diff
@@ -152,6 +161,131 @@ func (c *Client) ReadReleaseByTag(ctx context.Context, tkn string, repoURL url.U
 	return model.ToSvcGithubRelease(rls)
 }
 
+// TODO once svcmodel.Release has git tag field, separate gitTag parameter wont be needed
+func (c *Client) UpsertRelease(ctx context.Context, tkn string, repo svcmodel.GithubRepo, rls svcmodel.Release, gitTag string) error {
+	err := c.createRelease(ctx, tkn, repo, rls, gitTag)
+	if err != nil {
+		if errors.Is(err, errGithubReleaseAlreadyExists) {
+			err = c.updateRelease(ctx, tkn, repo, rls, gitTag)
+			if err != nil {
+				return fmt.Errorf("failed to update release: %w", err)
+			}
+		}
+
+		return fmt.Errorf("failed to create release: %w", err)
+	}
+
+	return nil
+}
+
+// TODO once svcmodel.Release has git tag field, separate gitTag parameter wont be needed
+func (c *Client) DeleteRelease(ctx context.Context, tkn string, repo svcmodel.GithubRepo, tagName string) error {
+	// Release can be deleted only by release ID
+	// Therefore I need to get release ID first
+	id, err := c.getReleaseIDByTag(ctx, tkn, repo, tagName)
+	if err != nil {
+		if errors.Is(err, errGithubReleaseNotFound) {
+			return svcerrors.NewGithubReleaseNotFoundError().Wrap(err)
```

Comment:
Yes. Exactly. 💪 
I think it's called idempotent operation. Can (and perhaps should) be a part of this endpoint documentation, so that FE isn't scared.

Reply:
Yes, I am going to update API documentation because now it is behind current implementation.